File: docs/snippets/python/benchmarking/simple_benchmark.py
```python
import time
import asyncio
from pathlib import Path
from kreuzberg import Kreuzberg, ExtractionConfig

async def benchmark_extractions():
    config = ExtractionConfig(use_cache=False)
    kreuzberg = Kreuzberg(config)
    file_path = "document.pdf"
    num_runs = 10

    start = time.perf_counter()
    for _ in range(num_runs):
        result = kreuzberg.extract_file(file_path)
    sync_duration = time.perf_counter() - start
    avg_sync = sync_duration / num_runs

    print(f"Sync extraction ({num_runs} runs):")
    print(f"  - Total time: {sync_duration:.3f}s")
    print(f"  - Average: {avg_sync:.3f}s per extraction")

    start = time.perf_counter()
    tasks = [kreuzberg.extract_file_async(file_path) for _ in range(num_runs)]
    await asyncio.gather(*tasks)
    async_duration = time.perf_counter() - start

    print(f"\nAsync extraction ({num_runs} parallel runs):")
    print(f"  - Total time: {async_duration:.3f}s")
    print(f"  - Average: {async_duration / num_runs:.3f}s per extraction")
    print(f"  - Speedup: {sync_duration / async_duration:.1f}x")

    cache_config = ExtractionConfig(use_cache=True)
    kreuzberg_cached = Kreuzberg(cache_config)

    print("\nFirst extraction (populates cache)...")
    start = time.perf_counter()
    result1 = await kreuzberg_cached.extract_file_async(file_path)
    first_duration = time.perf_counter() - start
    print(f"  - Time: {first_duration:.3f}s")

    print("Second extraction (from cache)...")
    start = time.perf_counter()
    result2 = await kreuzberg_cached.extract_file_async(file_path)
    cached_duration = time.perf_counter() - start
    print(f"  - Time: {cached_duration:.3f}s")
    print(f"  - Cache speedup: {first_duration / cached_duration:.1f}x")

if __name__ == "__main__":
    asyncio.run(benchmark_extractions())
```

File: docs/snippets/typescript/benchmarking/simple_benchmark.ts
```typescript
import { Kreuzberg, ExtractionConfig } from "kreuzberg";

async function benchmarkExtractions(): Promise<void> {
  const config: ExtractionConfig = { useCache: false };
  const kreuzberg = new Kreuzberg(config);
  const filePath = "document.pdf";
  const numRuns = 10;

  let start = performance.now();
  for (let i = 0; i < numRuns; i++) {
    await kreuzberg.extractFile(filePath);
  }
  const syncDuration = (performance.now() - start) / 1000;
  const avgSync = syncDuration / numRuns;

  console.log(`Sync extraction (${numRuns} runs):`);
  console.log(`  - Total time: ${syncDuration.toFixed(3)}s`);
  console.log(`  - Average: ${avgSync.toFixed(3)}s per extraction`);

  start = performance.now();
  const tasks = Array(numRuns)
    .fill(null)
    .map(() => kreuzberg.extractFile(filePath));
  await Promise.all(tasks);
  const asyncDuration = (performance.now() - start) / 1000;

  console.log(`\nAsync extraction (${numRuns} parallel runs):`);
  console.log(`  - Total time: ${asyncDuration.toFixed(3)}s`);
  console.log(`  - Average: ${(asyncDuration / numRuns).toFixed(3)}s per extraction`);
  console.log(`  - Speedup: ${(syncDuration / asyncDuration).toFixed(1)}x`);

  const cacheConfig: ExtractionConfig = { useCache: true };
  const kreuzbergCached = new Kreuzberg(cacheConfig);

  console.log("\nFirst extraction (populates cache)...");
  start = performance.now();
  const result1 = await kreuzbergCached.extractFile(filePath);
  const firstDuration = (performance.now() - start) / 1000;
  console.log(`  - Time: ${firstDuration.toFixed(3)}s`);

  console.log("Second extraction (from cache)...");
  start = performance.now();
  const result2 = await kreuzbergCached.extractFile(filePath);
  const cachedDuration = (performance.now() - start) / 1000;
  console.log(`  - Time: ${cachedDuration.toFixed(3)}s`);
  console.log(`  - Cache speedup: ${(firstDuration / cachedDuration).toFixed(1)}x`);
}

benchmarkExtractions().catch(console.error);
```
